### PR TITLE
Bug/invalid field count

### PR DESCRIPTION
**Pull request by mnelson4**
Addresses https://github.com/eventespresso/event-espresso-core/issues/418

### CODE REVIEW

**Comment by mnelson4** [comment]
## Testing
* [ ] verify this resolves https://github.com/eventespresso/event-espresso-core/issues/418

**Comment by mnelson4** [comment]
> Hey Mike I'm going to need some steps to test this partly because a user reported the issue and I'm afraid I'm in the dark about what this is supposed to resolve.

Oh sorry, I thought you knew more about it. I looked into it some more and the method with the bug in it isn't used by EE core anywhere, so they must have a custom code that's using this. 

Seeing how the custom code isn't used anywhere, we could probably skip manual testing. Or I could maybe add a unit test... that might be good...

**Comment by joshfeck** [comment]
A unit test might be good in case this happened to happen because of a merge conflict that went bad.

**Comment by youri--** [comment]
As for the custom code... Issue #431 describes a custom code example (see "Reproduce" section) of how this bug came to light (I reported #418 on the EE forum). I.e. using PHP to programmatically add a event_espresso_category to an event.

**Comment by mnelson4** [comment]
Ok I added a unit test that shows the issue was resolved. I think that will do in place of manual testing. @nerrad can you please re-review?

File: tests/testcases/core/db_classes/EE_CPT_Base_Test.php [hunk 1]
```diff
@@ -118,6 +118,24 @@ public function testGetPrettyCurrentPostRemainingTheSame()
...
+        $e = $this->new_model_obj_with_dependencies('Event');
+        /**
+         * @var $e EE_Event
+         */
+        $ee_term_taxonomy = $e->add_event_category(
+            'Rebel Ships',
+            'Events look at important Rebel Space Ships'
+        );
+        $this->assertTrue($ee_term_taxonomy instanceof EE_Term_Taxonomy);
+        $this->assertNotEquals(0, $ee_term_taxonomy);
```
Comment (nerrad) [inline tests/testcases/core/db_classes/EE_CPT_Base_Test.php]: Not sure both these assertions are necessary?  All that needs to be asserted is just the first right?

Comment (mnelson4) [reply tests/testcases/core/db_classes/EE_CPT_Base_Test.php]: oh oups, the second one should be testing the count

**Pull request merged**